packages/block-library/src/navigation-menu/index.php

```diff
@@ -94,10 +98,29 @@ function render_block_navigation_menu( $attributes, $content, $block ) {
  * @return string Returns  an HTML list from innerBlocks.
  */
 function build_navigation_menu_html( $block, $colors ) {
+	$on_hover_mode = ! empty( $attributes['menuEventActivation'] ) && 'onHover' === $attributes['menuEventActivation'];
+
 	$html = '';
+	$level = 0;
 	foreach ( (array) $block['innerBlocks'] as $key => $menu_item ) {
+		$has_sub_items = count( (array) $menu_item['innerBlocks'] ) > 0;
 
 		$html .= '<li class="wp-block-navigation-menu-item ' . $colors['bg_css_classes'] . '"' . $colors['bg_inline_styles'] . '>' .
+			(
+				( $has_sub_items && ! $on_hover_mode )
+					? (
+						'<input
+							id="toggle-handler-' . $level . '-' . $key . '"
+							class="wp-block-navigation-menu-item__toggle-handler"
+							type="checkbox"
+						>' .
+						'<label
```

Comment:
Perhaps take a look at [Bootstrap's "Dropdown" implementation](https://getbootstrap.com/docs/4.0/components/dropdowns/) for a11y guidance here.